README.md

```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
+				});
+			}
+		}
+	}
+}
+```
+
+Note that you are using `value` and not `get`. With `get`, every time `todos` changes, a brand new derived list will be generated. Since the derived list updates itself based on the source list, make sure only to instantiate the derived list once.
+
+## API
+
+The `can-derive` plugin adds the `dFilter` method to `can.List.prototype`, so all filtered lists are derived from a `can.List`.
+
+### dFilter
+
+`list.filter(predicate) -> DerivedList`
+
+Generates a derived list based on a predicate function.
+
+### attr
+
+`derivedList.attr() --> Array`
+
+Gets an array of all the elements in the derived list.
+
+`derivedList.attr(index) --> Object`
+
+Reads an element from an index on the filtered list. This references the *element in the source list* that *is at the specified index in the filtered list*.
+
+### each
+
+`derivedList.each(fn) --> DerivedList`
+
+Iterates through the DerivedList, calling a function for each element. The elements provided inside the function will be the elements *in the source list*. The filtered list will be updated only if the outcome of the predicate function changes.
+
+### Other can.List methods
+
+Since DerivedList is a [can.List](http://canjs.com/docs/can.List.html), the following methods are also available:
+
+- [filter](http://canjs.com/docs/can.List.prototype.filter.html)
+- [indexOf](http://canjs.com/docs/can.List.prototype.indexOf.html)
+- [map](http://canjs.com/docs/can.List.prototype.map.html)
+- [replace](http://canjs.com/docs/can.List.prototype.replace.html)
+- [slice](http://canjs.com/docs/can.List.prototype.slice.html)
+
+### Disabled can.List methods
+
+The filtered list is not changed manually, but is maintained as the source list changes. Because of this, the following `can.List` methods are disabled:
+
+- push
+- pop
+- shift
+- unshift
+- splice
+
+## Performance
+
+`can-derive` optimizes for indertions and removals, completing them in O(log n) time. This means that changes to the source list will automatically update the derived list in O(log n) time, compared to the standard O(n) time you would expect in other implementations.
 
 It does this by:
 
-- keeping `completed` in a RBTree.
-- listening to when any items are added or removed from the source list.
-- listening to when the predicate function value changes for any item.
+- Keeping the derived list in a red-black tree
+- Listening for additions or removals in the source list
+- Listening to when the result of the predicate function changes for any item
+
+This algorithm was originally discussed in [this StackExchange thread](http://cs.stackexchange.com/questions/43447/order-preserving-update-of-a-sublist-of-a-list-of-mutable-objects-in-sublinear-t/44502#44502).
+
+### When to Use
+
+In general, it is preferable to use `can-derive` over alternative approaches when:
+
+- Your source list contains 10 or more items
+- You would need to perform a `diff` operation in an alternative approach
```

Comment:
I don't think we should assume that the end-user knows a `diff` is occurring to qualify the use of can-derive. 

- You need to know how the filtered list changed, for instance when rendering in the DOM (which typically depends on a `diff`).